core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -284,6 +283,11 @@ private void decPendingAndFinishIfNeeded() {
 
     private void finish() {
         if (finished.compareAndSet(false, true)) {
+            try {
+                primary.updateGlobalCheckpoint();
```

Comment:
Thanks for the suggestion @ywelsch. I spent a lot of time looking into implementing this. This difficulty here is that we only want the background sync to fire if a replication operation has not recently broadcast the latest global checkpoint to the replica shards. During indexing, for example, we would not want to fire off a background sync if an imminent inline sync is going to carry the latest global checkpoint for us anyway. This means that the background sync needs to become aware of a lot of state: what was the last global checkpoint synced, and when was it synced. If the current global checkpoint is larger than the last synced global checkpoint and sufficient time has passed since the last sync, then we can fire off a background sync. I have a change sitting on my local disk that does this, but before I polish it enough to include in a PR, I would rather discuss with @bleskes if this is the path that we should go down. Either way, I would prefer that it not be included in this PR. I think this PR here is a reasonable size, the change we are talking about here sits on my disk at +219/-66, is tricky, is not polished, doesn't include changes so tests compile, and doesn't include tests for the new functionality. Thus, I want to keep it out of this PR no matter what we decide.